mailserver/mailserver.go

```diff
@@ -531,15 +544,28 @@ func (s *WMailServer) processRequestInBundles(
 
 	// Publish
 	for _, batch := range batches {
-		output <- batch
+		select {
+		case output <- batch:
+		// It might happen that during producing the batches,
+		// the connection with the peer goes down and
+		// the consumer of `output` channel exits prematurely.
+		// In such a case, we should stop pushing batches and exit.
+		case <-cancel:
+			log.Error("[mailserver:processRequestInBundles] failed to push all batches",
```

Comment:
I would make this a `Warn` or even an `Info`, it sounds like it is more or less a normal and expected scenario, as opposed to the timeout in this loop.

Reply:
That's true, it might be a perfectly normal scenario when a peer looses a connection to a mailserver, especially when being on a 3G or something like that.